Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py

```diff
@@ -6324,6 +6325,179 @@ def test_set_mirror_last_run_in_6_5(self, mocker):
             assert set_last_run.called is False
 
 
+class TestFetchWithLookBack:
+
+    LAST_RUN = {}
+    INCIDENTS = [
+        {
+            'incident_id': 1,
+            'created': (datetime.utcnow() - timedelta(hours=3)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 2,
+            'created': (datetime.utcnow() - timedelta(hours=1)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 3,
+            'created': (datetime.utcnow() - timedelta(minutes=29)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 4,
+            'created': (datetime.utcnow() - timedelta(minutes=19)).strftime('%Y-%m-%dT%H:%M:%S')
+        },
+        {
+            'incident_id': 5,
+            'created': (datetime.utcnow() - timedelta(minutes=9)).strftime('%Y-%m-%dT%H:%M:%S')
+        }
+    ]
+    
+    def example_fetch_incidents(self):
+
+        from CommonServerPython import get_fetch_run_time_with_look_back, look_for_incidents_in_last_run, get_incidents_from_response, \
+            set_next_fetch_run
+
+        incidents = []
+
+        params = demisto.params()
+        fetch_limit_param = params.get('limit')
+        look_back = int(params.get('look_back', 0))
+        first_fetch = params.get('first_fetch')
+        time_zone = params.get('time_zone', 0)
+        return_incidents_by_limit = params.get('return_incidents_by_limit', True)
+
+        last_run = demisto.getLastRun()
+        fetch_limit = last_run.get('limit') or fetch_limit_param
+
+        if not last_run.get('limit'):
+            last_run['limit'] = fetch_limit
+
+        incidents = look_for_incidents_in_last_run(last_run)
+        start_fetch_time, end_fetch_time = get_fetch_run_time_with_look_back(last_run, first_fetch, look_back, time_zone)
+
+        if incidents:
+            last_run, incidents = set_next_fetch_run(last_run, incidents, fetch_limit_param, start_fetch_time, end_fetch_time, look_back, 
+                                                    'created', 'incident_id')
+            demisto.setLastRun(last_run)
+            return incidents
+
+        query = self.build_query(start_fetch_time, end_fetch_time, fetch_limit, return_incidents_by_limit)
+        incident_res = self.get_incidents_request(query)
+
+        incidents = get_incidents_from_response(incident_res, last_run, 'incident_id')
+
+        new_last_run, incidents = set_next_fetch_run(last_run, incidents, fetch_limit_param, start_fetch_time, end_fetch_time, look_back,
+                                                    'created', 'incident_id')
+
```

Comment:
I think we can move `remove_duplicate_incidents_from_response` to be inside `set_next_fetch_run` since the incidents are processed there anyway. Then we can rename `set_next_fetch_run` to something like `handle_fetch_incidents_run`

Reply:
note that in this example you check if there are incidents in the last run but then you don't pass `save_incidents_in_last_run=True`, let's pass it in order to show exactly how it should be done

Reply:
I think it's better if we can make it more modular for the developer, WDYT?
Let's talk about it in our call